### ISSUE TITLE
orientdb process keeps database files open even after DISCONNECT

### ISSUE THREAD

**Issue by karel-un** [orientechnologies/orientdb#8158]
## Expected behavior  
After DISCONNECT from database in remote mode orientdb process should release file descriptors of given database files.

## Actual behavior  
After DISCONNECT from database in remote mode orientdb process has database files still open. This is inconsistent behaviour and prevents connecting to database from other process, from console in plocal mode or working with them. To fix this orientdb process has to be restarted.

## Steps to reproduce

1. start orientdb service (Gentoo example):
```
rc-service orientdb start
```
2. list open files of orientdb service and see that no files are open for any database:
```
lsof -p `ps ax | grep -v grep | grep orientdb | cut -d " " -f 1`
```
3. run orient db console, connect to database in remote mode and immediately disconnect:
```
localhost /tmp # orientdb

OrientDB console v.2.2.33 (build 77584cd6827f647cf4aa231cf27bd6f10bc04e2c, branch 2.2.x) https://www.orientdb.com
Type 'help' to display all the supported commands.
Installing extensions for GREMLIN language v.2.6.0

orientdb> connect remote:localhost/GratefulDeadConcerts admin admin

Connecting to database [remote:localhost/GratefulDeadConcerts] with user 'admin'...OK
orientdb {db=GratefulDeadConcerts}> disconnect

Disconnecting from the database [GratefulDeadConcerts]...OK
orientdb> quit
```
4. list open files of orientdb service again and see that GratefulDeadConcerts files are still open:
```
lsof -p `ps ax | grep -v grep | grep orientdb | cut -d " " -f 1`
```

**Comment by tglman** [comment]
Hi @karel-un,

Yes OrientDB will close the databases only at shutdown, this is an architectural decision to avoid a lot of load and unload of all the cache/metadatas of the database.

Regards

**Comment by karel-un** [comment]
Hi @tglman,

thanks for explanation. I would guess it is architectural decision. But anyway it is inconsistent behaviour and at least it would be nice to introduce some config file option to disable it. Disconnected database should not have its files still open.

Or to have some parameter for DISCONNECT command to actually release files too.

**Comment by tglman** [comment]
Hi @karel-un,

We may introduce in future a command to force close a database, but we do not consider this high priority, so it won't be in 3.0 it may be included in 3.1/3.2

Regards

**Comment by karel-un** [comment]
Ok, thanks for your help @tglman. I will leave issue open, feel free to close it as needed.
